Data/DynamicInputPack.cs

```diff
@@ -776,43 +782,51 @@ private void WriteGameID(string location)
+            foreach (HostDevice device in HostDevices)
             {
-                string destImagePath = Path.Combine(location, GetImageName(texture.TexturePath));
-
-                // Unlikely that we get here but skip textures that don't exist
-                if (!File.Exists(texture.TexturePath))
+                foreach (HostKey key in device.HostKeys)
                 {
-                    continue;
-                }
-
-                // Prevents the file from trying to overwrite itself.
-                if (texture.TexturePath == destImagePath)
-                {
-                    continue;
+                    //exports the images for the default_host_controls
+                    WriteImage(location, key);
                 }
-
-                const bool overwrite = true;
-                File.Copy(texture.TexturePath, destImagePath, overwrite);
             }
 
-            foreach (var device in _host_devices)
+            foreach (DynamicInputTexture texture in Textures)
             {
-                foreach (var key in device.HostKeys)
+                //exports the images for the output_textures
+                WriteImage(location, texture);
+                foreach (HostDevice device in texture.HostDevices)
                 {
-                    // skip if the file does not exist.
-                    if (!File.Exists(key.TexturePath))
-                        continue;
-
-                    const bool overwrite = true;
-                    var texture_location = GetHostDeviceKeyTextureLocation(location, device, key);
-                    Directory.CreateDirectory(Path.GetDirectoryName(texture_location));
-                    File.Copy(key.TexturePath, texture_location, overwrite);
+                    foreach (HostKey key in device.HostKeys)
+                    {
+                        //exports the images for the host_controls
+                        WriteImage(location, key);
+                    }
                 }
             }
         }
 
-        private void WriteHostControls(JsonWriter writer, ObservableCollection<HostDevice> hostDevices)
+        private void WriteImage(string location, Interfaces.ExportableImage image)
+        {
+            // Unlikely that we get here but skip textures that don't exist
+            if (!File.Exists(image.TexturePath))
+                return;
+
+            //check the output Path
+            image.RelativeTexturePath = CheckRelativeTexturePath(image);
+            string output_location = Path.Combine(location, image.RelativeTexturePath);
+
+            // Prevents the file from trying to overwrite itself.
+            if (output_location == image.TexturePath)
+                return;
+
+            //write the image
+            const bool overwrite = true;
+            Directory.CreateDirectory(Path.GetDirectoryName(output_location));
+            File.Copy(image.TexturePath, output_location, overwrite);
+        }
```

Comment:
to make things easier i created an interface for the images, so you don't need two seperate functions anymore.